src/protocols/vnc/vnc.h

```diff
@@ -133,6 +143,27 @@ typedef struct guac_vnc_client {
      * Clipboard encoding-specific writer.
      */
     guac_iconv_write* clipboard_writer;
+    
+    /**
+     * A lock that will be locked when retrieving required credentials from
+     * the client, and unlocked when credentials have been retrieved.
+     */
+    pthread_mutex_t vnc_credential_lock;
+    
+    /**
+     * A condition to use for signaling the thread when credentials have been
+     * retrieved from the client.
+     */
+    pthread_cond_t vnc_credential_cond;
+    
+    /**
+     * A field to track flags related to retrieving required credentials
+     * from the client.
```

Comment:
The nature of that relation should be documented. Are these flags the parameters that are required but have not yet been provided? Are they the parameters that have been provided so far?